crates/accelerate/src/commutation_cancellation.rs

```diff
@@ -0,0 +1,280 @@
+
+                            if num_qargs == 1 && Z_ROTATIONS.contains(&op_gate) {
+                                cancellation_sets
+                                    .entry(CancellationSetKey {
+                                        gate: GateOrRotation::ZRotation,
+                                        qubits: smallvec![wire],
+                                        com_set_index: com_set_idx,
+                                        second_index: None,
+                                    })
+                                    .or_insert_with(Vec::new)
+                                    .push(*node);
+                            }
+                            if num_qargs == 1 && X_ROTATIONS.contains(&op_gate) {
+                                cancellation_sets
+                                    .entry(CancellationSetKey {
+                                        gate: GateOrRotation::XRotation,
+                                        qubits: smallvec![wire],
+                                        com_set_index: com_set_idx,
+                                        second_index: None,
+                                    })
+                                    .or_insert_with(Vec::new)
+                                    .push(*node);
+                            }
+                            // Don't deal with Y rotation, because Y rotation doesn't commute with
+                            // CNOT, so it should be dealt with by optimized1qgate pass
+                            if num_qargs == 2
+                                && dag.get_qargs(instr.qubits).first().unwrap() == &wire
+                            {
+                                let second_qarg = dag.get_qargs(instr.qubits)[1];
+                                cancellation_sets
+                                    .entry(CancellationSetKey {
+                                        gate: GateOrRotation::Gate(op_gate),
+                                        qubits: smallvec![wire, second_qarg],
+                                        com_set_index: com_set_idx,
+                                        second_index: node_indices
+                                            .get(&(*node, Wire::Qubit(second_qarg)))
+                                            .copied(),
+                                    })
+                                    .or_insert_with(Vec::new)
+                                    .push(*node);
+                            }
+                        }
+                    }
+                })
+            }
+        }
+    });
+
+    for (cancel_key, cancel_set) in &cancellation_sets {
+        if cancel_set.len() > 1 {
+            if let GateOrRotation::Gate(g) = cancel_key.gate {
+                if SUPPORTED_GATES.contains(&g) {
+                    for &c_node in &cancel_set[0..(cancel_set.len() / 2) * 2] {
+                        dag.remove_op_node(c_node);
+                    }
+                }
```

Comment:
I think this needs a continue here because this is an xor with the other two gate fields. In the python this was if gate else if rotation.
```suggestion
                }
                continue;
```